lib/linklist.c

```diff
@@ -296,3 +297,36 @@ void list_add_list(struct list *l, struct list *m)
 	for (n = listhead(m); n; n = listnextnode(n))
 		listnode_add(l, n->data);
 }
+
+struct list *list_dup(struct list *l)
+{
+	struct list *new = list_new();
+	struct listnode *ln;
+	void *data;
+
+	for (ALL_LIST_ELEMENTS_RO(l, ln, data))
+		listnode_add(new, data);
+
+	return new;
+}
+
+void list_sort(struct list *list, int (*cmp)(const void **, const void **))
+{
+	struct listnode *ln, *nn;
+	int i = -1;
+	void *data;
+	size_t n = list->count;
+	void *items[n];
```

Comment:
If we are declaring data on the stack, should we have a limit?  What happens if the linked list is several million items long?

Reply:
fair